app/src/main/java/co/infinum/rxpokemon/ui/login/LoginPresenter.java

```diff
@@ -28,31 +48,66 @@ public void init() {
     @Override
     public void login(String username, String password) {
 
+        cancel();
         view.setState(new LoginViewState(State.LOADING));
 
         LoginParams params = new LoginParams(username, password);
 
-        interactor.loginUser(params, new Listener<LoginResponse>() {
-            @Override
-            public void onSuccess(LoginResponse result) {
+        loginDisposable = interactor.loginUser(params)
+                .subscribeOn(ioScheduler)
+                .observeOn(mainThreadScheduler)
+                .retry(NUMBER_OF_RETRIES)
+                .subscribeWith(new DisposableObserver<LoginResponse>() {
+                    @Override
+                    public void onNext(LoginResponse loginResponse) {
+                        User.getInstance().setEmail(loginResponse.getEmail());
```

Comment:
Slažem se. Ima na par mjesta u projektu sličnih gluposti. Ako će biti vremena budem to sredio kako treba prije radionice.